### ISSUE TITLE
Calculation of energy distribution losses

### ISSUE THREAD

**Issue by wouterterlouw** [quintel/refinery#40]
At the moment, the distribution converter of carriers with distribution losses cannot be calculated, because import/export of the converter is dependent on the loss of the converter, which is dependent on the import/export of the converter. Since the losses vary per year and country the solution would be to have refinery calculate the output efficiency for these converters.

The general situation is:
![screenshot_8_9_13_9_39_am](https://f.cloud.github.com/assets/4319656/936885/df60587a-00c6-11e3-9d36-bda280691b3c.jpg)

The supply, import, demand and export are in the same carrier group. The loss is in another carrier group.

The converter is balanced by the following equation:

```
demand + export + loss = supply + import
   D   +   E    +  L   =   S    +   I
```

where D = demand, E = export, L = loss, S = supply, I = import.

Loss is calculated using a `output.loss` efficiency and is specific for country and year.

**What do we know (in refinery!)?**
- demand
- supply

**What additional data do we have?**
- loss from the energy balance: `-EB(losses, electricity)`

**What do we need?**
- we need to ensure that the `output.loss` efficiency matches the ratio of total converter demand and loss

**Ideas**

The efficiencies can be calculated using:

```
output.coal efficiency = coal output / converter demand
                          D + E       D + E 
output.coal efficiency = ------- = -----------
                          S + I     D + E + L
```

When `supply < demand + loss`, import will occur and the export = 0 and the second equation applies: `output.coal efficiency = D / (D+L)`

When `supply > demand + loss`, export will occur and import = 0 and the first equation applies: `output.coal efficiency = (D + E) / S` or `(D + E) / (D + E + L)`.

A short term solution is to calculate the demand and supply by hand (adding all known demands and supply streams using refinery output, calculate the corresponding efficiency, set this efficiency and set the loss demand. In this way the converter will be calculated. But every time anything changes in the analysis we need to do this calculation again. Automating the calculation in the analysis is non-trivial because of the many demand links to these converters that are determined over many different Excel analyses.

Is it possible to create a special kind of class of converters that Refinery can calculate the loss conversion for?

This issue is relevant for:
- energy_power_hv_network_electricity
- energy_heat_network_steam_hot_water
- energy_distribution_crude_oil
- energy_national_gas_network_natural_gas
